ie_serving/models/ir_engine.py

```diff
@@ -140,9 +123,98 @@ def set_keys(self, mapping_config):
         else:
             return self._set_names_in_config_as_keys(mapping_data)
 
-    def infer(self, data: dict, batch_size=None):
-        if batch_size is not self.net.batch_size and self.batch_size == 0:
-            self.net.batch_size = batch_size
+    def infer(self, data: dict):
+        try:
+            results = self.exec_net.infer(inputs=data)
+        except Exception as e:
+            message = "Error occurred during inference execution: {}".format(
+                str(e))
+            logger.debug("[Model: {}, version: {}] --- {}".format(
+                self.model_name, self.model_version, message))
+            return None, message
+        return results, None
+
+    def scan_input_shapes(self, data: dict):
+        #   Takes dictionary of input_name:numpy_array pairs.
+        reshape_required = False
+        inputs_shapes = {}
+        for input_name, input_data in data.items():
+            net_input_shape = tuple(self.net.inputs[input_name].shape)
+            inputs_shapes[input_name] = input_data.shape
+            if net_input_shape != input_data.shape:
+                logger.debug("[Model: {}, version: {}] --- Shape change "
+                             "required for input: {}. Current "
+                             "shape: {}. Required shape: {}"
+                             .format(self.model_name, self.model_version,
+                                     input_name, net_input_shape,
+                                     input_data.shape))
+                reshape_required = True
+        return reshape_required, inputs_shapes

```

Comment:
The same situation, if inputs_shapes dict is empty then reshape is not required.